doc/centos-tutorial.txt

```diff
@@ -0,0 +1,141 @@
+  # yumdownloader --source xz-devel
+  # rpm2cpio xz-5.2.2-1.el7.src.rpm | cpio -idv
+  # tar xf xz-5.2.2.tar.gz
+  # cd xz-5.2.2
+  # patch -Np1 < ../xz-5.2.2-compat-libs.patch
+  # patch -Np1 < ../xz-5.2.2-man-page-day.patch
+  # ./configure --enable-static
+  # make
+  # cp ./src/liblzma/.libs/liblzma.a /usr/lib64/
+  # cd ~
+
+2. Installing the MANA directory and compiling MANA
+
+  $ git clone https://github.com/mpickpt/mana.git
+  $ cd mana
+  $ git checkout feature/centos
+  
+Here, we need to make some changes to the configure-mana file - these changes
+will eventually be included in a new pull request and merged into this branch.
+For now, it is included as a patch in the root directory.
+
+  $ git apply configure-mana.patch
+  $ ./configure-mana
+  $ make -j mana
+
+This should build MANA smoothly.
+
+B. LAUNCHING, CHECKPOINTING AND RESTARTING A MPI APPLICATION
+
+We'll first add the bin directory to PATH. Ensure that these commands are done
+from within the root directory of MANA.
+
+  $ echo 'export PATH=$PATH:{PATH TO MANA}/bin' >> ~/.bashrc
+  $ source ~/.bashrc
+
+By default, this should be /home/{YOUR USERNAME}/mana.
+
+1. Launching an MPI application
+
+The MANA directory comes with many test MPI applications that can be found in
+contrib/mpi-proxy-plugin/test. Depending on the application, we may require
+more than 1 copy of the process running - for example, ping_pong.mana.exe
+requires 2. To support this, change the argument after -np accordingly. For this
+tutorial, we'll use mpi_hello_world.mana.exe, which only requires 1 copy
+running.
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_launch \ 
+      contrib/mpi-proxy-split/test/mpi_hello_world.mana.exe
+
+If the application is launched properly, you should see the following printed:
+
+  Hello world from processor test, rank 0 out of 1 processors
+  Will now sleep for 500 seconds ...
+
+From here, we can either exit the program (through CTRL-C), or continue on to
+the next set of instructions to checkpoint and restart the application.
+
+2. Checkpointing and restarting a MPI application
+
+Currently, the fix for mana_restart has not been pushed to the feature/centos
+branch yet; it's going through some testing in the final phases, but we'll
+simply grab the pull request and patch it in.
+
+  $ wget https://github.com/mpickpt/mana/pull/29.patch
+  $ git apply 29.patch
+  $ make -j mana
+
+Here, we want to either open another terminal, or have the launched MPI
+application be open in the background. The following demonstrates the latter.
+
+If an application is already launched, we do CTRL-Z, followed by:
+
+  $ bg
+
+Otherwise, we'll launch an application the same way as above, but as a
+background process (note the &).
+
+  $ mana_coordinator
+  $ mpirun -np 1 mana_launch \ 
+      contrib/mpi-proxy-split/test/mpi_hello_world.mana.exe &
+
+Subsequently, we only need the following command to checkpoint the application.
+
+  $ mana_status --checkpoint
```

Comment:
Let's add a statement here that there are two other ways to specify when to checkpoint.  See `man mana` for details, where the manpage for MANA can be found in `DMTCP_ROOT/manpages`.  The second way is interval checkpointing (`-i`) and the third way is to modify the MPI application so that one and only one MPI process uses the mechanism outlined in `DMTCP_ROOT/test/plugin/applic-initiated-ckpt`.